spacy/gold/corpus.py

```diff
@@ -54,6 +64,40 @@ def walk_corpus(path: Union[str, Path]) -> List[Path]:
                 locs.append(path)
         return locs
 
+    def __call__(self, nlp: "Language", loc: Union[Path, str],) -> Iterator[Example]:
+        """Yield examples from the data.
+
+        nlp (Language): The current nlp object.
+        loc (Path): The file or directory to read from.
+        YIELDS (Example): The examples.
+
+        DOCS: https://spacy.io/api/corpus#train_dataset
+        """
+        loc = util.ensure_path(loc)
+        ref_docs = self.read_docbin(nlp.vocab, self.walk_corpus(loc))
+        if self.gold_preproc:
+            examples = self.make_examples_gold_preproc(nlp, ref_docs)
+        else:
+            examples = self.make_examples(nlp, ref_docs, self.max_length)
+        yield from examples
+
+    def count(self, nlp: "Language", loc) -> int:
+        """Returns count of words in train examples.
+
+        nlp (Language): The current nlp. object.
+        RETURNS (int): The word count.
+
+        DOCS: https://spacy.io/api/corpus#count_train
```

Comment:
```suggestion
        DOCS: https://spacy.io/api/corpus#count
```

Reply:
We could also consider removing this method, as we'd rather not ask user-provided readers to supply it.

Reply:
Ah, I thought we needed this for the word counts. But it looks like we're currently only using it in one test?

I'm definitely in favour of removing stuff we don't need and keeping the API simple.